src/sentry/api/serializers/models/release.py

```diff
@@ -243,7 +283,15 @@ def serialize(self, obj, attrs, user, *args, **kwargs):
             'authors': attrs.get('authors', []),
             'projects': attrs.get('projects', [])
         }
-        if attrs['tag']:
+        first_and_last_seen = attrs.get('first_and_last_seen')
+        if first_and_last_seen:
+            d.update(
+                {
+                    'firstEvent': first_and_last_seen['first_seen'],
+                    'lastEvent': first_and_last_seen['last_seen'],
+                }
+            )
+        elif attrs['tag']:
```

Comment:
Wouldn't it make more sense for `get_attrs` to always return a `first_seen` and `last_seen` key, instead of the `first_and_last_seen` and `tag` structures that contain that same data? Similarly, if the return types of `__get_release_data_no_environment` and `__get_release_data_with_environment` were equivalent (making the methods basically substitutable depending on the parameters passed to `get_attrs`, it'd remove the need for the conditional on line 253 and make this generally simpler/easier to understand. (Or am I overlooking something?)

Reply:
Good idea. I think that was just how it was done before, and I wasn't thinking. :)